### ISSUE TITLE
SQLConnectInfo issue with environment variables and @

### ISSUE THREAD

**Issue by tony-kerz** [proftpd/proftpd#507]
### What I Did
```
SQLConnectInfo %{env:DB_DATABASE}@%{env:DB_HOST} %{env:DB_USER} %{env:DB_PASSWORD}
```
### What I Expected/Wanted

the `%{env:<blah>}` directives to be substituted

 > apparently only `DB_USER` (and likely `DB_PASSWORD`) are being picked up, but `DB_DATABASE` and `DB_HOST` aren't being honored. i suspect it has something to do with the `@` in between...

i see the following in the sql log file:
```
2017-05-13 20:20:01,517 mod_sql/4.3[7]: entering 	mysql cmd_defineconnection
2017-05-13 20:20:01,517 mod_sql/4.3[7]:   name: 'default'
2017-05-13 20:20:01,517 mod_sql/4.3[7]:   user: 'user1'
2017-05-13 20:20:01,517 mod_sql/4.3[7]:   host: 'localhost'
2017-05-13 20:20:01,517 mod_sql/4.3[7]:     db: ''
2017-05-13 20:20:01,518 mod_sql/4.3[7]:   port: '3306'
2017-05-13 20:20:01,518 mod_sql/4.3[7]:    ttl: '0'
2017-05-13 20:20:01,518 mod_sql/4.3[7]: exiting 	mysql cmd_defineconnection
2017-05-13 20:20:01,518 mod_sql/4.3[7]: connection 'default' successfully established
```
> we can see that `DB_USER` which is set to `user1` is being picked up successfully

> just verified that setting `DB_DATABASE=db@host` works fine and seems like a decent workaround, but submitting this anyway in case anyone else gets stumped and/or you want to fix the `@` thing 

### ProFTPD Version and Configuration

`ProFTPD 1.3.6 (stable)`

`./configure --with-modules=mod_sql:mod_sql_mysql:mod_tls:mod_sftp`

**Comment by Castaglia** [comment]
Thanks for the bug report!  This should now be fixed in master; the fix has also been backported to the 1.3.6 branch.

**Comment by Castaglia** [comment]
I think you may need to use the ProFTPD 1.3.6a release; that's the maintenance release which contains the fix in question.  It was not part of the ProFTPD 1.3.6 release; the backport/fix was to the 1.3.6 _branch_, and the next release from that branch, containing the fix, was ProFTPD 1.3.6a.

What does `proftpd -V` show for you?
